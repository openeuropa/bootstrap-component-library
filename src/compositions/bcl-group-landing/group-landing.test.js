import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import demoData from "@openeuropa/bcl-group-landing/data";

const template = "@oe-bcl/bcl-group-landing/group-landing.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Group Landing", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
});
