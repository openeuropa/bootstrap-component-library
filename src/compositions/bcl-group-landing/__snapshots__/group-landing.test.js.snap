// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Group Landing renders correctly 1`] = `
<jest>
  <main>
    <div
      class="container"
    >
      <div
        class="mt-4 mt-md-5"
      >
        <img
          alt="alt img"
          class="d-none d-md-block w-100"
          src="https://picsum.photos/seed/1002/1100/255"
        />
        <div
          class="p-md-3-5 pt-3 bg-lighter d-md-flex n-mx-container"
        >
          <h2
            class="mb-0 d-inline-block fw-bold"
          >
            Group Name
          </h2>
          <div
            class="d-grid d-md-block ms-auto"
          >
            <a
              class="d-md-inline-block me-2 p-2 d-none"
              href="/example.html"
            >
              Leave group
            </a>
            <button
              aria-expanded="false"
              autocomplete="off"
              class="d-inline-block mt-3 mt-md-0 mb-4 mb-md-0 dropdown-toggle btn btn-primary btn-md"
              data-bs-toggle="dropdown"
              id="dropdown-1"
              type="button"
            >
              Create content
            </button>
            <ul
              aria-labelledby="dropdown-1"
              class="dropdown-menu"
            >
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  I'm a link
                </a>
              </li>
              <li>
                <button
                  class="dropdown-item"
                  type="button"
                >
                  I'm a button
                </button>
              </li>
              <li>
                <hr
                  class="dropdown-divider"
                />
              </li>
              <li>
                <button
                  aria-disabled="true"
                  class="dropdown-item"
                  disabled="true"
                  type="button"
                >
                  I'm a disabled button
                </button>
              </li>
              <li>
                <button
                  class="dropdown-item active"
                  type="button"
                >
                  I'm a active button
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <nav
        class="bcl__nav bg-gray n-mx-container nav nav-tabs"
      >
        <a
          class="nav-link active"
          href="/example.html"
        >
          Home
        </a>
        <a
          class="nav-link"
          href="/example.html"
        >
          Members
        </a>
        <a
          class="nav-link"
          href="/example.html"
        >
          Content
        </a>
        <a
          aria-expanded="false"
          class="nav-link dropdown-toggle"
          data-bs-toggle="dropdown"
          href="#"
          id="dropdown-1"
          role="button"
        >
          Links
        </a>
        <ul
          aria-labelledby="dropdown-1"
          class="dropdown-menu"
        >
          <li>
            <a
              class="dropdown-item"
              href="/example.html"
            >
              I'm a link
            </a>
          </li>
          <li>
            <button
              class="dropdown-item"
              type="button"
            >
              I'm a button
            </button>
          </li>
          <li>
            <hr
              class="dropdown-divider"
            />
          </li>
          <li>
            <button
              aria-disabled="true"
              class="dropdown-item"
              disabled="true"
              type="button"
            >
              I'm a disabled button
            </button>
          </li>
          <li>
            <button
              class="dropdown-item active"
              type="button"
            >
              I'm a active button
            </button>
          </li>
        </ul>
        <a
          class="nav-link"
          href="/example.html"
        >
          Link
        </a>
      </nav>
      <a
        aria-controls="banner-description"
        aria-expanded="false"
        class="w-100 text-center py-4  d-block d-md-none"
        data-bs-toggle="collapse"
        href="#banner-description"
        role="button"
      >
        <span
          class="text-not-collapsed"
        >
          See group detail
        </span>
        <span
          class="text-collapsed"
        >
          Hide group detail
        </span>
        <svg
          class="ms-2-5 bi icon--xs"
        >
          <use
            xlink:href="#caret-down-fill"
          />
        </svg>
      </a>
      <div
        class="collapse d-md-block"
        id="banner-description"
      >
        <div
          class="row mt-md-4 mb-md-5 pb-md-5"
        >
          <div
            class="col-md-8"
          >
            <div
              class="ps-md-3"
            >
              <div
                class="d-block"
              >
                <p
                  class="mb-0 me-4 d-inline-block text-muted"
                >
                  <svg
                    class="me-2 align-top mt-1 bi icon--xs"
                  >
                    <use
                      xlink:href="#key-fill"
                    />
                  </svg>
                  Public group
                </p>
                <p
                  class="mb-0 me-4 d-inline-block text-muted"
                >
                  <svg
                    class="me-2 align-top mt-1 bi icon--xs"
                  >
                    <use
                      xlink:href="#people-fill"
                    />
                  </svg>
                  22 Members
                </p>
              </div>
              <p
                class="mt-3"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse volutpat ultricies massa, a dapibus libero porta nec. Sed facilisis dictum vestibulum. Fusce commodo hendrerit diam, pretium tempus leo varius sit amet. Etiam interdum, orci at sagittis luctus, lorem libero tempus mauris, a fermentum libero orci semper lacus. Duis tristique fringilla magna, eu egestas dolor molestie non.
              </p>
            </div>
          </div>
          <div
            class="col-md-4"
          >
            <h4
              class="mb-3 pb-2"
            >
              Contact information
            </h4>
            <div
              class="d-flex mb-3"
            >
              <div
                class="flex-shrink-0"
              >
                <img
                  alt="Stefan Mayer profile picture"
                  class="rounded-pill"
                  src="https://picsum.photos/50/50"
                />
              </div>
              <div
                class="flex-grow-1 ms-3"
              >
                <a
                  class="fs-5 text-underline-hover mb-0"
                  href="/example.html"
                >
                  Stefan Mayer
                </a>
                <div
                  class="d-flex flex-wrap"
                >
                  <span
                    class="text-muted text-nowrap me-4-5"
                  >
                    DG Test
                  </span>
                  <span
                    class="text-muted text-nowrap"
                  >
                    Director
                  </span>
                </div>
              </div>
            </div>
            <div
              class="d-flex mb-3"
            >
              <div
                class="flex-shrink-0"
              >
                <img
                  alt="Stefan Mayer profile picture"
                  class="rounded-pill"
                  src="https://picsum.photos/50/50"
                />
              </div>
              <div
                class="flex-grow-1 ms-3"
              >
                <a
                  class="fs-5 text-underline-hover mb-0"
                  href="/example.html"
                >
                  Stefan Mayer
                </a>
                <div
                  class="d-flex flex-wrap"
                >
                  <span
                    class="text-muted text-nowrap me-4-5"
                  >
                    DG Test
                  </span>
                  <span
                    class="text-muted text-nowrap"
                  >
                    Director
                  </span>
                </div>
              </div>
            </div>
            <div
              class="d-flex mb-3"
            >
              <div
                class="flex-shrink-0"
              >
                <img
                  alt="Stefan Mayer profile picture"
                  class="rounded-pill"
                  src="https://picsum.photos/50/50"
                />
              </div>
              <div
                class="flex-grow-1 ms-3"
              >
                <a
                  class="fs-5 text-underline-hover mb-0"
                  href="/example.html"
                >
                  Stefan Mayer
                </a>
                <div
                  class="d-flex flex-wrap"
                >
                  <span
                    class="text-muted text-nowrap me-4-5"
                  >
                    DG Test
                  </span>
                  <span
                    class="text-muted text-nowrap"
                  >
                    Director
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row mt-3 mt-md-5 pt-md-5"
      >
        <div
          class="col-md-8"
        >
          <h4
            class="mb-4 pb-2"
          >
            Recent activities
          </h4>
          <div>
            <div
              class="d-flex bg-lighter py-3 px-3 px-md-3-5 mb-3-5"
            >
              <div>
                <img
                  class="rounded-circle me-3-5"
                  src="https://picsum.photos/50/50?random=1"
                />
              </div>
              <div>
                <p
                  class="mb-2"
                >
                  <a
                    class="fw-bold text-underline-hover"
                    href="/example.html"
                  >
                    Adjur Tichar
                  </a>
                   commented on the article 
                  <a
                    class="text-underline-hover"
                    href="#"
                  >
                    SpaceX satellites spotted: how good is this for our health and environment?
                  </a>
                </p>
                <p
                  class="mb-0"
                >
                  <svg
                    class="me-2 align-top mt-1 bi icon--xs"
                  >
                    <use
                      xlink:href="#clock"
                    />
                  </svg>
                  8 hours ago
                </p>
              </div>
            </div>
            <div
              class="d-flex bg-lighter py-3 px-3 px-md-3-5 mb-3-5"
            >
              <div>
                <img
                  class="rounded-circle me-3-5"
                  src="https://picsum.photos/50/50?random=2"
                />
              </div>
              <div>
                <p
                  class="mb-2"
                >
                  <a
                    class="fw-bold text-underline-hover"
                    href="/example.html"
                  >
                    Adjur Tichar
                  </a>
                   commented on the article 
                  <a
                    class="text-underline-hover"
                    href="#"
                  >
                    SpaceX satellites spotted: how good is this for our health and environment?
                  </a>
                </p>
                <p
                  class="mb-0"
                >
                  <svg
                    class="me-2 align-top mt-1 bi icon--xs"
                  >
                    <use
                      xlink:href="#clock"
                    />
                  </svg>
                  8 hours ago
                </p>
              </div>
            </div>
            <div
              class="d-flex bg-lighter py-3 px-3 px-md-3-5 mb-3-5"
            >
              <div>
                <img
                  class="rounded-circle me-3-5"
                  src="https://picsum.photos/50/50?random=3"
                />
              </div>
              <div>
                <p
                  class="mb-2"
                >
                  <a
                    class="fw-bold text-underline-hover"
                    href="/example.html"
                  >
                    Adjur Tichar
                  </a>
                   commented on the article 
                  <a
                    class="text-underline-hover"
                    href="#"
                  >
                    SpaceX satellites spotted: how good is this for our health and environment?
                  </a>
                </p>
                <p
                  class="mb-0"
                >
                  <svg
                    class="me-2 align-top mt-1 bi icon--xs"
                  >
                    <use
                      xlink:href="#clock"
                    />
                  </svg>
                  8 hours ago
                </p>
              </div>
            </div>
            <div
              class="d-flex bg-lighter py-3 px-3 px-md-3-5 mb-3-5"
            >
              <div>
                <img
                  class="rounded-circle me-3-5"
                  src="https://picsum.photos/50/50?random=4"
                />
              </div>
              <div>
                <p
                  class="mb-2"
                >
                  <a
                    class="fw-bold text-underline-hover"
                    href="/example.html"
                  >
                    Steve Di Marzio
                  </a>
                   published the article 
                  <a
                    class="text-underline-hover"
                    href="/example.html"
                  >
                    SpaceX satellites spotted: how good is this for our health and environment?
                  </a>
                   in 
                  <a
                    class="text-underline-hover"
                    href="/example.html"
                  >
                    News
                  </a>
                </p>
                <p
                  class="mb-0"
                >
                  <svg
                    class="me-2 align-top mt-1 bi icon--xs"
                  >
                    <use
                      xlink:href="#clock"
                    />
                  </svg>
                  8 hours ago
                </p>
              </div>
            </div>
            <div
              class="d-flex bg-lighter py-3 px-3 px-md-3-5 mb-3-5"
            >
              <div>
                <img
                  class="rounded-circle me-3-5"
                  src="https://picsum.photos/50/50?random=5"
                />
              </div>
              <div>
                <p
                  class="mb-2"
                >
                  <a
                    class="fw-bold text-underline-hover"
                    href="/example.html"
                  >
                    Adjur Tichar
                  </a>
                   commented on the article 
                  <a
                    class="text-underline-hover"
                    href="/example.html"
                  >
                    SpaceX satellites spotted: how good is this for our health and environment?
                  </a>
                </p>
                <p
                  class="mb-0"
                >
                  <svg
                    class="me-2 align-top mt-1 bi icon--xs"
                  >
                    <use
                      xlink:href="#clock"
                    />
                  </svg>
                  8 hours ago
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-md-4"
        >
          <h4
            class="mt-4 mt-md-0 mb-4 pb-2"
          >
            Latest members
          </h4>
          <div
            class="d-flex mb-3"
          >
            <div
              class="flex-shrink-0"
            >
              <img
                alt="Stefan Mayer profile picture"
                class="rounded-pill"
                src="https://picsum.photos/50/50?random=6"
              />
            </div>
            <div
              class="flex-grow-1 ms-3"
            >
              <a
                class="fs-5 text-underline-hover mb-0"
                href="/example.html"
              >
                Stefan Mayer
              </a>
              <div
                class="d-flex flex-wrap"
              >
                <span
                  class="text-muted text-nowrap me-4-5"
                >
                  DG Test
                </span>
                <span
                  class="text-muted text-nowrap"
                >
                  Director
                </span>
              </div>
            </div>
          </div>
          <div
            class="d-flex mb-3"
          >
            <div
              class="flex-shrink-0"
            >
              <img
                alt="Stefan Mayer profile picture"
                class="rounded-pill"
                src="https://picsum.photos/50/50?random=7"
              />
            </div>
            <div
              class="flex-grow-1 ms-3"
            >
              <a
                class="fs-5 text-underline-hover mb-0"
                href="/example.html"
              >
                Stefan Mayer
              </a>
              <div
                class="d-flex flex-wrap"
              >
                <span
                  class="text-muted text-nowrap me-4-5"
                >
                  DG Test
                </span>
                <span
                  class="text-muted text-nowrap"
                >
                  Director
                </span>
              </div>
            </div>
          </div>
          <div
            class="d-flex mb-3"
          >
            <div
              class="flex-shrink-0"
            >
              <img
                alt="Stefan Mayer profile picture"
                class="rounded-pill"
                src="https://picsum.photos/50/50?random=8"
              />
            </div>
            <div
              class="flex-grow-1 ms-3"
            >
              <a
                class="fs-5 text-underline-hover mb-0"
                href="/example.html"
              >
                Stefan Mayer
              </a>
              <div
                class="d-flex flex-wrap"
              >
                <span
                  class="text-muted text-nowrap me-4-5"
                >
                  DG Test
                </span>
                <span
                  class="text-muted text-nowrap"
                >
                  Director
                </span>
              </div>
            </div>
          </div>
          <div
            class="d-flex mb-3"
          >
            <div
              class="flex-shrink-0"
            >
              <img
                alt="Stefan Mayer profile picture"
                class="rounded-pill"
                src="https://picsum.photos/50/50?random=9"
              />
            </div>
            <div
              class="flex-grow-1 ms-3"
            >
              <a
                class="fs-5 text-underline-hover mb-0"
                href="/example.html"
              >
                Stefan Mayer
              </a>
              <div
                class="d-flex flex-wrap"
              >
                <span
                  class="text-muted text-nowrap me-4-5"
                >
                  DG Test
                </span>
                <span
                  class="text-muted text-nowrap"
                >
                  Director
                </span>
              </div>
            </div>
          </div>
          <div
            class="d-flex mb-3"
          >
            <div
              class="flex-shrink-0"
            >
              <img
                alt="Stefan Mayer profile picture"
                class="rounded-pill"
                src="https://picsum.photos/50/50?random=10"
              />
            </div>
            <div
              class="flex-grow-1 ms-3"
            >
              <a
                class="fs-5 text-underline-hover mb-0"
                href="/example.html"
              >
                Stefan Mayer
              </a>
              <div
                class="d-flex flex-wrap"
              >
                <span
                  class="text-muted text-nowrap me-4-5"
                >
                  DG Test
                </span>
                <span
                  class="text-muted text-nowrap"
                >
                  Director
                </span>
              </div>
            </div>
          </div>
          <div
            class="d-flex mb-3"
          >
            <div
              class="flex-shrink-0"
            >
              <img
                alt="Stefan Mayer profile picture"
                class="rounded-pill"
                src="https://picsum.photos/50/50?random=11"
              />
            </div>
            <div
              class="flex-grow-1 ms-3"
            >
              <a
                class="fs-5 text-underline-hover mb-0"
                href="/example.html"
              >
                Stefan Mayer
              </a>
              <div
                class="d-flex flex-wrap"
              >
                <span
                  class="text-muted text-nowrap me-4-5"
                >
                  DG Test
                </span>
                <span
                  class="text-muted text-nowrap"
                >
                  Director
                </span>
              </div>
            </div>
          </div>
          <a
            class="py-2 d-inline-block"
            href="/example.html"
          >
            See all group members
            <svg
              class="ms-2-5 bi icon--xs"
            >
              <use
                xlink:href="#chevron-right"
              />
            </svg>
          </a>
        </div>
      </div>
      <h4
        class="my-4 pt-3 pb-2"
      >
        Latest contributions
      </h4>
      <div
        class="mb-4 bcl-listing bcl-listing--highlight-3-col"
      >
        <div
          class="gy-4-5 row row-cols-1 row-cols-md-3 g-2"
        >
          <div
            class="col"
          >
            <div
              class="listing-item--highlight border-0 bg-lighter h-100 rounded-2 card"
            >
              <img
                alt="alt img"
                class="card-img-top"
                src="https://picsum.photos/seed/1002/600/400"
              />
              <div
                class="card-body"
              >
                <div
                  class="mb-3"
                >
                  <span
                    class="me-2 badge bg-primary"
                  >
                    Label
                  </span>
                  <span
                    class="badge bg-primary"
                  >
                    Label
                  </span>
                </div>
                <h5
                  class="card-title"
                >
                  <a
                    class="text-underline-hover"
                    href="/example"
                  >
                    Card title
                  </a>
                </h5>
                <p
                  class="card-text mb-2"
                >
                  Some quick example text to build on the card and title and make up the bulk of the cards content.
                </p>
                <div
                  class="mb-4 mt-4"
                >
                  <span
                    class="text-muted me-3 text-nowrap"
                  >
                    8 hours ago
                  </span>
                  <span
                    class="text-muted me-3 text-nowrap"
                  >
                    1 comment
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="listing-item--highlight border-0 bg-lighter h-100 rounded-2 card"
            >
              <img
                alt="alt img"
                class="card-img-top"
                src="https://picsum.photos/seed/1003/600/400"
              />
              <div
                class="card-body"
              >
                <div
                  class="mb-3"
                >
                  <span
                    class="me-2 badge bg-primary"
                  >
                    Label
                  </span>
                  <span
                    class="me-2 badge bg-primary"
                  >
                    Label
                  </span>
                  <span
                    class="badge bg-primary"
                  >
                    Label
                  </span>
                </div>
                <h5
                  class="card-title"
                >
                  <a
                    class="text-underline-hover"
                    href="/example"
                  >
                    Card title
                  </a>
                </h5>
                <p
                  class="card-text mb-2"
                >
                  Some quick example text to build on the card and title and make up the bulk of the cards content.
                </p>
                <div
                  class="mb-4 mt-4"
                >
                  <span
                    class="text-muted me-3 text-nowrap"
                  >
                    8 hours ago
                  </span>
                  <span
                    class="text-muted me-3 text-nowrap"
                  >
                    1 comment
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="listing-item--highlight border-0 bg-lighter h-100 rounded-2 card"
            >
              <img
                alt="alt img"
                class="card-img-top"
                src="https://picsum.photos/seed/1004/600/400"
              />
              <div
                class="card-body"
              >
                <div
                  class="mb-3"
                >
                  <span
                    class="badge bg-primary"
                  >
                    Label
                  </span>
                </div>
                <h5
                  class="card-title"
                >
                  <a
                    class="text-underline-hover"
                    href="/example"
                  >
                    Card title
                  </a>
                </h5>
                <p
                  class="card-text mb-2"
                >
                  Some quick example text to build on the card and title and make up the bulk of the cards content.
                </p>
                <div
                  class="mb-4 mt-4"
                >
                  <span
                    class="text-muted me-3 text-nowrap"
                  >
                    8 hours ago
                  </span>
                  <span
                    class="text-muted me-3 text-nowrap"
                  >
                    1 comment
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <a
          class="mb-4-5 pt-1 pb-2 mt-3-5 d-block"
          href="/example.html"
        >
          See all group contributions
          <svg
            class="ms-2-5 bi icon--xs"
          >
            <use
              xlink:href="/icons.svg#chevron-right"
            />
          </svg>
        </a>
      </div>
    </div>
  </main>
</jest>
`;
